### DISCUSSION TITLE
How to decorate a default template?

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#16892] [Q&A] [answered]
I want to cross out an entire row in a `DataGrid` based on a property in the view model behind that row.
Assuming I can apply a style to a row based on the view model, I am experimenting with how to add UI elements to the row. Here is my attempt:

```xaml
<Style Selector="DataGridRow">
  <Setter Property="Template">
    <ControlTemplate>
      <Border Background="Cyan" Padding="0 4">
        <ContentPresenter />
      </Border>
    </ControlTemplate>
  </Setter>
</Style>
```

But the only thing drawn on the row is the Border control.
How can I get the original template to draw itself in the designated position?

**Comment by thevortexcloud** [comment]
In this situation you are overriding the entire template. You can modify the template without overriding it if you use a regular style. Otherwise you will need to copy and paste the original template, as most built in controls need specific things in them for them to work properly.

I suspect you will have a better time using a style with a class binding.

https://docs.avaloniaui.net/docs/guides/data-binding/binding-classes

**Reply by AArnott** [reply]
Thanks. I use class bindings a lot, and they're great. But in this case I don't `DataGridRow` isn't an element that I ever use directly, so there is nowhere to apply the `Classes.SomeClass` attribute to. 
And even if there were, how does that allow me to modify the template to _add_ a control to draw a line through the entire row?

**Reply by thevortexcloud** [reply, marked as answer]
Ah, you are right. Apologies, I seem to have confused myself. Looking at the existing template, you can probably abuse some of the shapes that are there already if you don't mind things looking a bit different.

https://github.com/AvaloniaUI/Avalonia/blob/77fe0400fc57139c4a53fc0c7d148fca36c80cdd/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L317-L358

That said since you seem to be trying to add a border, there is already a border on the row that is set up for template bindings. Which means you can target it via a style.

You can modify the existing template with the `/template/` selector. 

https://docs.avaloniaui.net/docs/0.10.x/styling/selectors#template

But if you really need to add a new control inside the template, then you will need to override the template. There is not really a simple way to avoid that.

**Reply by AArnott** [reply]
Thank you. 

FYI The Border control was just one way I was trying to strikeout text. I don't want to displace any Border control already in use.
